Gossip DataColumnSidecar decoding uses highest-supported-milestone schema — all Fulu column sidecars rejected when Gloas is scheduled

### Summary

On networks that have Gloas scheduled (observed on `plataberget` / glamsterdam-devnet-8, `GLOAS_FORK_EPOCH: 1536`), teku decodes incoming `data_column_sidecar_{subnet_id}` gossip with the **Gloas** `DataColumnSidecar` schema while the chain is still in **Fulu**. As a result, every gossiped column sidecar fails deserialization — on our devnet this is ~20 rejected messages/sec on every teku node, continuously since genesis (2026-08-13), on both `ethpandaops/teku:glamsterdam-devnet-8` and `ethpandaops/teku:master` builds.

```
Caused by: tech.pegasys.teku.infrastructure.ssz.sos.SszDeserializeException: First variable element offset doesn't match the end of fixed part
    at tech.pegasys.teku.infrastructure.ssz.schema.impl.AbstractSszContainerSchema.sszDeserializeTree(AbstractSszContainerSchema.java:484)
    ...
WARN  - Failed to decode gossip message on topic /eth2/5c943807/data_column_sidecar_4/ssz_snappy, raw message: 0xe48801040400...
```

### The rejected messages are valid Fulu sidecars on the correct Fulu topic

Snappy-decompressing a sample raw message from the log yields a fully self-consistent **Fulu**-shape `DataColumnSidecar`:

- `index = 4` (matches topic subnet), declared uncompressed length 17508 matches exactly
- offsets `356 / 16740 / 17124` → first offset **356** = the Fulu fixed part (`8 + 3×4 + SignedBeaconBlockHeader(208) + inclusion proof(4×32)`)
- column = 16384 bytes = 8 cells × 2048, commitments = proofs = 384 = 8 × 48 (an 8-blob block)
- sane `signed_block_header` (slot 28341, proposer 15411 — a live lighthouse node's head-slot proposal)

And the topic digest is the *current Fulu* one, not Gloas: fulu base digest `127d903d` XOR `sha256(epoch_le64 ‖ max_blobs_le64)[:4]` for the active blob params `(0, 21)` = **`5c943807`**. (The Gloas digest would be `98c910cf`.) So the publisher is conformant; the decode side is not.

### Root cause

`DataColumnSidecarSubnetSubscriptions` resolves the decode schema from the *highest supported milestone* instead of the fork the subscription instance belongs to:

https://github.com/Consensys/teku/blob/4d86366d17e7be9562b571745af160764ff76117/networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/subnets/DataColumnSidecarSubnetSubscriptions.java#L65-L69

```java
final SpecVersion specVersion =
    spec.forMilestone(spec.getForkSchedule().getHighestSupportedMilestone());
this.dataColumnSidecarSchema =
    SchemaDefinitionsFulu.required(specVersion.getSchemaDefinitions())
        .getDataColumnSidecarSchema();
```

Since #9929 changed the Gloas container (removed `kzg_commitments`, `signed_block_header`, `kzg_commitments_inclusion_proof`; added `slot`, `beacon_block_root`), the Gloas fixed part is 56 bytes vs Fulu's 356 — so with `GLOAS_FORK_EPOCH` set to anything non-far-future, the Fulu-era topic handler expects offset 56 and rejects every valid Fulu message with offset 356. On networks without Gloas scheduled the highest milestone *is* Fulu, which is why this doesn't reproduce elsewhere.

The constructor already receives the per-fork `forkInfo` (used correctly for the topic name/digest), so the fix looks like a one-liner:

```java
final SpecVersion specVersion = spec.atEpoch(forkInfo.getFork().getEpoch());
```

This also does the right thing for the BPO subscription variants (`atEpoch` maps a BPO activation epoch to its base milestone) and for `GossipForkSubscriptionsGloas`, which inherits this code path with a Gloas `forkInfo`.

Note: `ExecutionProofSubnetSubscriptions` (L67) has the same highest-milestone pattern — latent for now, but it will break the same way the first time that container changes across a scheduled fork.

### Impact

- Gossip DAS is effectively disabled for teku nodes pre-Gloas on such networks; they stay in sync only via req/resp column fetch (which selects schema correctly), so the failure is easy to miss.
- Peers publishing valid sidecars are presumably penalized by gossip scoring.
- The bug silently "self-heals" at the Gloas transition, making it look transient.

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -16,5 +16,6 @@
  - Improved debug/beacon/states endpoint to allow searching of the finalized state root, to assist third party products searching on roots.
 
 ### Bug Fixes
+ - Fixed `data_column_sidecar` gossip decoding to use the schema of the topic's fork instead of the highest supported milestone. Previously, on networks with Gloas scheduled, every Fulu-era column sidecar received via gossip failed deserialization.
  - Fixed a regression where archive nodes using `leveldb-tree` storage would take an extremely long time to start up.
  - Post-Electra, the `committee_index` query parameter in `GET /eth/v1/validator/attestation_data` is now ignored instead of rejected when non-zero, matching the behaviour of other consensus clients.

--- a/networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/subnets/DataColumnSidecarSubnetSubscriptions.java
+++ b/networking/eth2/src/main/java/tech/pegasys/teku/networking/eth2/gossip/subnets/DataColumnSidecarSubnetSubscriptions.java
@@ -62,8 +62,7 @@ public class DataColumnSidecarSubnetSubscriptions extends CommitteeSubnetSubscri
     this.debugDataDumper = debugDataDumper;
     this.forkInfo = forkInfo;
     this.forkDigest = forkDigest;
-    final SpecVersion specVersion =
-        spec.forMilestone(spec.getForkSchedule().getHighestSupportedMilestone());
+    final SpecVersion specVersion = spec.atEpoch(forkInfo.getFork().getEpoch());
     this.dataColumnSidecarSchema =
         SchemaDefinitionsFulu.required(specVersion.getSchemaDefinitions())
             .getDataColumnSidecarSchema();

